src/cmd/pay.rs

```diff
@@ -125,18 +135,54 @@ fn print_txn(
 pub struct Payee {
     address: PublicKey,
     amount: Hnt,
+    memo: u64,
```

Comment:
Is 0 an actual sentinel for a not-a-memo, or should this be an `Option<u64>`?

Reply:
The proto defines the memo as a u64, not an Option<u64>, so it needs to be set to _something_. AFAIK, protos with default values don't actually add any size to the serialization, setting it to 0 saves space (and makes the transaction cheapest).

Reply:
that's what I suspected. It's a little infuriating that protobuf v3 conflates `not set` and `defualt`

Reply:
I suppose we could use the syntax:
```
Txn {
            field_i_set: set,   
            ..Default::default() 
        }
```
 we probably should actually?

Reply:
if @madninja is okay with the divergence from current practice of setting the defaults by hand, I'd be for it

Reply:
I don’t have an opinion either way. I was mainly curious if not using Option was due to some implementation problem

Reply:
I'd shrug at the complexity judgement. We're looking at:

```
let txn = Txn {
   field_i_set: set,
   ...Default::default() 
};

if let Some(optional_value) =  optional_value {
  txn.optional_field = optional_value 
}
```

as opposed to

```
let txn = Txn {
   field_i_set: set,
   optional_field: if let Some(optional_value) =  optional_value {
      txn.optional_field = optional_value
   } else {
      default_value
   },
};
```

I'd argue the former is perhaps more readable and expressive (given that `default_value` in the second would literally be the default value, such as `0`, nothing that says `default`). But I think arguing over one or the other is a bit of a nit and not worth losing sleep over. I'd rather just maintain the current style.